0016-pre-shared-key/0016-pre-shared-key.md

```diff
@@ -0,0 +1,277 @@
+# Pre-Shared Key
+
+The Pre-Shared Key (PSK) protocol is an end-to-end transport protocol, used by the sender and receiver of an ILP payment to decide on a condition and fulfillment for a payment. By default, the protocol also encrypts any additional data sent along with the payment, using [AES-256-GCM](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard). The PSK data is authenticated through an [HMAC-SHA-256](https://en.wikipedia.org/wiki/Hash-based_message_authentication_code) which is used to generate the fulfillment of a PSK payment. The entirety of the PSK data, including public headesrs, encrypted private headers, and encrypted private data, is expected to be encoded into an octet-stream that forms the data portion of the [ILP Packet](https://github.com/interledger/rfcs/blob/master/0003-interledger-protocol/0003-interledger-protocol.md).
```

Comment:
Might be better to link to https://en.wikipedia.org/wiki/Galois/Counter_Mode for AES-256-GCM.

Also it should say "The full ILP payment is authenticated through..." (the PSK data is now authenticated with GCM)

"...and encrypted private data, is ~~expected to be~~ encoded..."